src/main/resources/static/api.yaml

```diff
@@ -2830,34 +2829,54 @@ components:
       type: object
       properties:
         name:
-          type: string
-          example: Raj Sharma
+          $ref: '#/components/schemas/PatientName'
         username:
           type: string
           example: raj.sharma@ncg
         password:
           type: string
-        yearOfBirth:
-          type: integer
-          example: 2000
+        dateOfBirth:
+          $ref: '#/components/schemas/DateOfBirth'
         gender:
           $ref: '#/components/schemas/Gender'
       required:
         - name
         - username
         - password
         - gender
+    PatientName:
+      type: object
+      properties:
+        firstName:
```

Comment:
since you are already introducing this under name attribute - is it required to have "firstName" and so on? 
- you can just call them as "first", "middle", "last". Also maybe you can consider - "given", "family", "middle"?

Reply:
Done